scripts/go-wrapper.sh

```diff
@@ -1,31 +1,77 @@
 #!/usr/bin/env bash
 #
-# To be used as a replacement for the `go` toolchain. The wrapper
-# unifies building binaries normally and building them with coverage
-# support using `go test -c`.
+# To be used with Goreleaser as `gobinary` implementation as a
+# replacement for the `go` toolchain.
+#
+# First function: prebuilt binaries
+#
+# The wrapper detects and returns prebuilt binaries to skip actual go
+# builds. If a binary `-o goreleaser/../some-binary` is requested but
+# `goreleaer-prebuilt/../some-binary` already exists, the prebuilt
+# binary is copied instead of building.
+#
+# Second function: coverage-enabled builds for Pulumi CLI
+#
+# This builds binaries via `go test -c` workaround. Disabled for
+# Windows builds. Only enabled on the Pulumi CLI binaries.
 
 PULUMI_TEST_COVERAGE_PATH=$PULUMI_TEST_COVERAGE_PATH
 
 set -euo pipefail
 
 PKG=github.com/pulumi/pulumi/pkg/v3/...
 SDK=github.com/pulumi/pulumi/sdk/v3/...
-COVERPKG=$PKG,$SDK
-
-if [ -z "$PULUMI_TEST_COVERAGE_PATH" ]; then
-    go "$@"
-else
-    case $1 in
-        build)
-            shift
-            go test -c -cover -coverpkg $COVERPKG "$@"
-            ;;
-        install)
-            echo "install command is not supported, please use build"
-            exit 1
-            ;;
-        *)
-            go "$@"
-            ;;
-    esac
-fi
+COVERPKG="$PKG,$SDK"
+
+case "$1" in
+    build)
```

Comment:
yuck, this wrapping business. motivation makes sense but I hope you don't mind if I still go a little "eugh" at having to do this.